frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -75,35 +74,14 @@ end
 
 ReaderBookmark.onPhysicalKeyboardConnected = ReaderBookmark.registerKeyEvents
 
-function ReaderBookmark:genItemTextMenuItem(type, get_string)
-    local text_type = {
-        text = _("highlighted text"),
-        all  = _("highlighted text and note"),
-        note = _("note, or highlighted text"),
-    }
-    if get_string then
-        return text_type[type]
-    end
-    return {
-        text = text_type[type],
-        checked_func = function()
-            return self.items_text == type
-        end,
-        callback = function()
-            self.items_text = type
-            G_reader_settings:saveSetting("bookmarks_items_text_type", type)
-        end,
-    }
-end
-
 function ReaderBookmark:addToMainMenu(menu_items)
     menu_items.bookmarks = {
         text = _("Bookmarks"),
         callback = function()
             self:onShowBookmark()
         end,
     }
-    if not Device:isTouchDevice() and not ( Device:hasScreenKB() or Device:hasSymKey() ) then
+    if not Device:isTouchDevice() and not (Device:hasScreenKB() or Device:hasSymKey()) then
```

Comment:
Yes, but there are no other examples of `a and (b or c)`. I am not really bothered about it, i only wrote `a and ( b or c )` because it looks (to me at least) easier to read but if you don't agree, i'll use `a and (b or c)` next time.

Reply:
Not strictly true, I wrote this. :-)

```
apps/reader/modules/readerfont.lua
143:        if not (Device:hasScreenKB() or Device:hasSymKey()) then
```

You could also write that as `not and not` but I kept it the same as vice versa in favor of greppability:
```
$ rg -F "Device:hasScreenKB() or Device:hasSymKey()"
apps/reader/modules/readerbookmark.lua
106:    if not Device:isTouchDevice() and not ( Device:hasScreenKB() or Device:hasSymKey() ) then

apps/reader/modules/readerlink.lua
236:    if Device:hasScreenKB() or Device:hasSymKey() then

apps/reader/modules/readerfont.lua
143:        if not (Device:hasScreenKB() or Device:hasSymKey()) then

apps/reader/readerui.lua
529:        if Device:hasScreenKB() or Device:hasSymKey() then

apps/reader/modules/readerrolling.lua
119:    if Device:hasScreenKB() or Device:hasSymKey() then
```

> because it looks (to me at least) easier to read but if you don't agree, i'll use a and (b or c) next time.

Generally speaking the code doesn't add spaces unless it's necessary for legibility, which may never be the case for a single level of parentheses. I had seen it but decided to ignore it.

Reply:
so the question is, which is more legible 

1. `not Device:isTouchDevice() and not (Device:hasScreenKB() or Device:hasSymKey())` or
2. `not Device:isTouchDevice() and not ( Device:hasScreenKB() or Device:hasSymKey() )`

Reply:
For this case, I think 1).
I would only add spaces when there are more, to make the nested ones stand out, ie.:
```
if a and (b or c) then
if a and ( (b and c) or (b and d) ) then

```

Reply:
What @poire-z said

Reply:
Understood. 👍